catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DurableFileEndpoint.java

```diff
@@ -0,0 +1,189 @@
+                        }
+                        return dir.exists();
+                    }
+
+                    @Override
+                    public boolean retrieveFile(String name, Exchange exchange)
+                            throws GenericFileOperationFailedException {
+                        exchange.setOut(exchange.getIn());
+                        return true;
+                    }
+
+                    @Override
+                    public void releaseRetreivedFileResources(Exchange exchange)
+                            throws GenericFileOperationFailedException {
+
+                    }
+
+                    @Override
+                    public boolean storeFile(String name, Exchange exchange)
+                            throws GenericFileOperationFailedException {
+                        return false;
+                    }
+
+                    @Override
+                    public String getCurrentDirectory() throws GenericFileOperationFailedException {
+                        return null;
+                    }
+
+                    @Override
+                    public void changeCurrentDirectory(String path)
+                            throws GenericFileOperationFailedException {
+
+                    }
+
+                    @Override
+                    public void changeToParentDirectory()
+                            throws GenericFileOperationFailedException {
+
+                    }
+
+                    @Override
+                    public List<EventfulFileWrapper> listFiles()
+                            throws GenericFileOperationFailedException {
+                        return null;
+                    }
+
+                    @Override
+                    public List<EventfulFileWrapper> listFiles(String path)
+                            throws GenericFileOperationFailedException {
+                        return null;
+                    }
+                });
+    }
+
+    @Override
+    public GenericFileProducer<EventfulFileWrapper> createProducer() throws Exception {
+        return null;
+    }
+
+    @Override
+    public Exchange createExchange(GenericFile file) {
+        Exchange exchange = createExchange();
+        if (file != null) {
+            file.bindToExchange(exchange);
+        }
+        return exchange;
+    }
+
+    @Override
+    public String getScheme() {
+        return "durable";
+    }
+
+    @Override
+    public char getFileSeparator() {
+        return File.separatorChar;
+    }
+
+    @Override
+    public boolean isAbsolute(String name) {
+        return Paths.get(name)
+                .isAbsolute();
+    }
+
+    @Override
+    protected String createEndpointUri() {
+        return file.toURI()
+                .toString();
+    }
+
+    public void setFile(File file) {
+        this.file = file;
+        // update configuration as well
+        getConfiguration().setDirectory(FileUtil.isAbsolute(file) ?
+                file.getAbsolutePath() :
```

Comment:
🤣 I do, I just haven't gotten them all replaced yet. Thanks for the reminder though. 😸